utils/vlog/yosys/json.py

```diff
@@ -0,0 +1,161 @@
+#!/usr/bin/env python3
+"""
+This is intended to provide a range of helper functions around the output of
+Yosys' `write_json`. Depending on the tasks, this may need to be flattened
+and/or in AIG format. In any case, at minimum `proc` and usually `prep` should
+be used before outputting the JSON.
+"""
+
+import os, sys
+import json
+
+class YosysModule:
+    def __init__(self, name, module_data):
+        self.name = name
+        self.data = module_data
+
+    # Return the name of the module
```

Comment:
We don't use the getter / setter paradigm in Python. If you want values to be "private" prefix them with `_`. You can also use the [`@property decorator`](https://www.python-course.eu/python3_properties.php) to make "generated properties", see example below;
```python

class YosysModule:

    @property
    def ports(self):
        plist = []
        for port, pdata in self.data["ports"].items():
            plist.append((port, len(pdata["bits"]), pdata["direction"]))
        return plist

y = YosysModule()
print(y.ports)

```